### DISCUSSION TITLE
Why duckDB choose to implement its own parquet reader?

### DISCUSSION THREAD

**Discussion by mingmwang** [duckdb/duckdb#2762] [Q&A]
Why duckDB choose to implement its own parquet reader instead of using the official c++ parquet readers? Is there any benchmark ?

**Comment by hannes** [comment]
I'm not aware of any official C++ reader? Which reader are you referring to? What's the reasoning behind the question, is anything not working for you?

**Comment by FrancoisLepoutre** [comment]
Hi Hannes,

Sorry if I am hijacking this thread in any but I have started to explore Duckdb in tandem with pyarrow. 

The duckdb-arrow combo is quite great! We are feeding the duck either straight through the duckdb feeding interface via the pyarrow reader from a prospective proprietary encrypted data source. Both feeding have their use. And we like this quite a lot :-)

I also expected that there where some code sharing between the pyarrow and duckdb interface since there was this:
https://arrow.apache.org/docs/cpp/api/formats.html#cpp-api-parquet

Both interfaces feed the duck equally well! Thanks for the great work!
